src/java/main/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -1428,7 +1401,7 @@ void testableCloseSocket() throws IOException {
 
         public boolean tunnelAuthInProgress() {
             // 1. SASL client is disabled.
-            if (!ZooKeeperSaslClient.isEnabled()) {
+            if (!zooKeeper.getClientConfig().isSaslClientEnabled()) {
```

Comment:
If {{clientConfig}} is {{final}} then you can replace this by {{clientConfig.isSaslClientEnabled()}}